drivers/unix/dir_access_unix.cpp

```diff
@@ -126,37 +126,34 @@ String DirAccessUnix::get_next() {
 
 	if (!dir_stream)
 		return "";
-	dirent *entry;
 
-	entry = readdir(dir_stream);
+	dirent *entry = readdir(dir_stream);
 
 	if (entry == NULL) {
-
 		list_dir_end();
 		return "";
 	}
 
-	//typedef struct stat Stat;
-	struct stat flags;
-
 	String fname = fix_unicode_name(entry->d_name);
 
-	String f = current_dir.plus_file(fname);
-
-	if (stat(f.utf8().get_data(), &flags) == 0) {
+	// Look at d_type to determine if the entry is a directory, unless
+	// its type is unknown (the file system does not support it) or if
+	// the type is a link, in that case we want to resolve the link to
+	// known if it points to a directory. stat() will resolve the link
+	// for us.
+	if (entry->d_type == DT_UNKNOWN ||
+		entry->d_type == DT_LNK) {
```

Comment:
Style issue here, our continuation indent width is 2 tabs.

Reply:
We don't wrap lines at 72 or 80 chars though, so in this case the condition is short enough to be on one line iMO.

Reply:
Thank you for the information about continuation indent style, and I will put the whole condition on one single line as you suggest. As it is a small issue, would you like me to squash the next commit before updating my branch?

Reply:
You can use `git commit --amend` to edit the commit, and then force push. Or if you committed already, you'd have to squash the two commits together indeed.